R/gh_rate_limit.R

```diff
@@ -20,6 +20,11 @@
 
 gh_rate_limit <- function(response = NULL, .token = NULL, .api_url = NULL, .send_headers = NULL) {
   if (is.null(response)) {
+    lifecycle::deprecate_warn(
```

Comment:
I think we can use `deprecate_soft()`, and also, is there a way to only emit a message instead of a warning? I don't think we need to remove this function, it is effectively  zero maintenance, and it can potentially break people's workflows. But the new function is better, so we could point people to that.

Reply:
FWIW github has deprecated that the `rate` component of the `rate_limit` API: https://docs.github.com/en/rest/rate-limit?apiVersion=2022-11-28 and I don't think there's much cost in forwarding people to it. But I don't feel that strongly and I could just remove the deprecation if you don't like it.

Reply:
In that case we could just take the info from `resources$core` instead, I guess. And switch to `deprecate_soft()`?

Reply:
It's not that easy to make that work because the return value actually quite different — `gh_rate_limits()` gives you a run down of all the rate limits used by GH in a data frame where `gh_rate_limit()` gives you the specifics for a request with a list.

Reply:
Or maybe make it superseded, and use `signal_stage()`?

Reply:
In that case we can keep it as is, and make it superseded? I doubt that GH will remove it any time soon.

Reply:
I don't really care either, so I'll just leave as is for now.